components/config-mgmt-service/integration_test/node_export_test.go

```diff
@@ -689,3 +689,84 @@ func TestNodeExportLoopBug(t *testing.T) {
 		})
 	}
 }
+
+func TestNodeExportPaging(t *testing.T) {
+	esClient := elastic.New(elasticsearchUrl)
+	c := config.New(esClient)
+	server := grpcserver.NewCfgMgmtServer(c)
+
+	lis := bufconn.Listen(1024 * 1024)
+	s := grpc.NewServer()
+	api.RegisterCfgMgmtServer(s, server)
+
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			t.Fatalf("Server exited with error: %v", err)
+		}
+	}()
+
+	dialer := func(string, time.Duration) (net.Conn, error) { return lis.Dial() }
+
+	conn, err := grpc.DialContext(context.Background(), "bufnet", grpc.WithDialer(dialer), grpc.WithInsecure())
+	defer conn.Close()
+	require.NoError(t, err)
+
+	client := api.NewCfgMgmtClient(conn)
+
+	names := []string{
+		"Taco Bell",
+		"taco bell",
+		"TACO BELL",
+		"tACO bELL",
+		"TACO bell",
+	}
+
+	// page size is hardcoded to 100, so to test the pagination code we need more than that.
+	numberOfNodes := 150
+	nodes := make([]iBackend.Node, numberOfNodes)
+	for i := 0; i < numberOfNodes; i++ {
+		nodes[i].Exists = true
+		nodes[i].NodeInfo = iBackend.NodeInfo{
+			NodeName:   names[i%len(names)],
+			EntityUuid: newUUID(),
+		}
+	}
+
+	// Add nodes with projects
+	suite.IngestNodes(nodes)
+	defer suite.DeleteAllDocuments()
+
+	response, err := client.NodeExport(context.Background(), &request.NodeExport{
+		OutputType: "json",
+		Sorting: &request.Sorting{
+			Field: backend.Name,
+		},
+	})
+
+	require.NoError(t, err)
+	require.NotNil(t, response)
+
+	data := make([]byte, 0)
+	for {
+		tdata, err := response.Recv()
+		if err != nil && err == io.EOF {
+			data = append(data, tdata.GetContent()...)
+			break
+		}
+
+		require.NoError(t, err)
+		data = append(data, tdata.GetContent()...)
+	}
+
+	actualNumberOfNodes := 0
+	dec := json.NewDecoder(bytes.NewReader(data))
+	for dec.More() {
+		var nodes []interface{}
+		err := dec.Decode(&nodes)
+		require.NoError(t, err)
+
+		actualNumberOfNodes += len(nodes)
+	}
+
+	assert.Equal(t, numberOfNodes, actualNumberOfNodes)
+}
```

Comment:
While debugging this code, I changed the hardcoded page size to 1 and every test calling `NodeExport` would fail because they only look at the first page of data. That is fairly safe to do since these tests use a lot fewer nodes than would be necessary to trigger pagination behavior, but it was still fairly confusing and now this test works differently in how it parses the result.